libdnf/dnf-repo.c

```diff
@@ -1215,7 +1339,13 @@ dnf_repo_check_internal(DnfRepo *repo,
         if (!dnf_repo_set_timestamp_modified(repo, error))
             return FALSE;
         age_of_data =(g_get_real_time() - priv->timestamp_modified) / G_USEC_PER_SEC;
-        if (age_of_data > permissible_cache_age) {
+        metadata_expire = dnf_repo_get_metadata_expire(repo);
+        if(metadata_expire != 0)
+            valid_time_allowed = metadata_expire <= permissible_cache_age ? metadata_expire  : permissible_cache_age;
```

Comment:
hmm, I would say indirectly? Currently dnf_context_set_cache_age will refresh the context object's cache age, which is then passed in as permissible cache age as an argument.( but not the metadata_expire from each repo )

Then the permissible cache age is  compared to the metadata_expire attribute for each repo(if those exist), and a lower one will be taken and checked against expiry. I took the suggestion from issue,  #252.

Do you think it is better that whenever dnf_context_set_cache_age() is called, we set the metadata_expire for every repo to be changed as well? I have not added it yet, but I am definitely ok for adding it if needed :D.